packages/react-native-session-replay/android/src/main/kotlin/com/datadog/reactnative/sessionreplay/utils/ReflectionUtils.kt

```diff
@@ -0,0 +1,36 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.reactnative.sessionreplay.utils
+
+import java.lang.reflect.Field
+
+internal class ReflectionUtils {
+    internal fun getDeclaredField(instance: Any, fieldName: String): Any? {
+        val className = instance.javaClass
+        val declaredField = searchForField(className, fieldName) ?: return null
+
+        declaredField.let {
+            it.isAccessible = true
+            return it.get(instance)
```

Comment:
I've removed the try/catch for the FieldNotDefinedException, since this can no longer happen. However, there's a question whether this `get` should be wrapped. At least three out of the four exceptions that it can throw are I believe not possible (NPR because we pass a not null object, access because we're explicitly setting the access previously, and IllegalArgumentException since we know that the declared field is on the instance), but that still leaves ExceptionInInitializerError. I'm not certain what would cause this to throw, so perhaps we should wrap it to be safe?

Reply:
We shouldn't catch `Error` instances. This means that JVM is completely broken, it cannot be recovered.

Reply:
alright, so I'll leave this block without try/catch for now